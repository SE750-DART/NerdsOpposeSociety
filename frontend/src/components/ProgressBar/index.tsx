import React from "react";
import styles from "./style.module.css";

interface ProgressBarProps {
  playersChosen: number;
  playersTotal: number;
}

const ProgressBar = (ProgressBarProps: ProgressBarProps) => {
  let { playersChosen } = ProgressBarProps;
  const { playersTotal } = ProgressBarProps;

  if (playersChosen > playersTotal) {
    playersChosen = playersTotal;
  }

  const percentageComplete = (playersChosen / playersTotal) * 100;
  const message = `${playersChosen}/${playersTotal} players have chosen`;

  return (
    <>
      <div className={styles.barContainer}>
        <div
          className={styles.barComplete}
          style={{ width: `${percentageComplete}%` }}
        />
      </div>

      <h5>{message}</h5>
    </>
  );
};

export default ProgressBar;
