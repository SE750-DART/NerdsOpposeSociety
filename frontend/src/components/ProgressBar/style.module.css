.barContainer {
  background-color: #ffffff;
  border-radius: 8px;
  box-shadow: inset 0 0 10px rgba(0, 0, 0, 0.25);
  height: 16px;
  width: 100%;
  margin-bottom: 12px;
}
.barComplete {
  background-color: #84b5de;
  border-radius: 8px;
  height: 100%;
}
