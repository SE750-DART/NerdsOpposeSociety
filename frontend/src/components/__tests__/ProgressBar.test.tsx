import React from "react";
import renderer from "react-test-renderer";
import ProgressBar from "../ProgressBar";

it("Progress bar snapshot testing", () => {
  const treeBar = renderer
    .create(<ProgressBar playersChosen={2} playersTotal={4} />)
    .toJSON();
  expect(treeBar).toMatchSnapshot();
});
