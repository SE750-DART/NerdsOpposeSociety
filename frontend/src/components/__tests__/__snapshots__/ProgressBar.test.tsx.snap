// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress bar snapshot testing 1`] = `
Array [
  <div
    className="barContainer"
  >
    <div
      className="barComplete"
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>,
  <h5>
    2/4 players have chosen
  </h5>,
]
`;
